### PR TITLE
Mm 43423 inconsistent hover icons

### PR DESCRIPTION
**Pull request by trilopin**
#### Summary
Make icon hover style consistent with https://zeroheight.com/29be2c109/p/00b099-icon-buttons .
dotmenu now works the following way:
- grey when no active, no hover
- grey darker background when hover
- blue when active (dropdown opened, no matter if hover)
- blue with darker background when active & hover

In the case of RHS filter, I considered active when:
- the dropdown is openend
- the dropdown is closed but we have selected an option different than default (all choices selected by clicking in "all" or by clicking any choice except "all")

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-43423

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/dot_menu.tsx [hunk 1]
```diff
@@ -114,6 +117,7 @@ const DotMenu = (props: DotMenuProps) => {
         <MenuButton
             title={props.title}
             ref={rootRef}
+            isActive={(props.isActive ?? false) || isOpen}
```
Comment (agarciamontoro) [inline webapp/src/components/dot_menu.tsx:120]: Isn't this the same as
```suggestion
            isActive={props.isActive ?? isOpen}
```

**Review by agarciamontoro — APPROVED** [review]
Looks good, @trilopin! Thank you :tada:

**Review by gbochora — APPROVED** [review]
Cool!

File: webapp/src/components/dot_menu.tsx (hunk 1 above)
Comment (gbochora) [reply webapp/src/components/dot_menu.tsx:120]: It seems these conditions are not the same. i.e. if `props.isActive` is `false` and `isOpen` is `true`, first condition result is `true`, and second is `false`.

Comment (agarciamontoro) [reply webapp/src/components/dot_menu.tsx:120]: Aaaah, that's true, because the ?? only checks for real `null`ity, right?

Comment (gbochora) [reply webapp/src/components/dot_menu.tsx:120]: Yep

**Review by abhijit-singh — APPROVED** [review]
Awesome!

**Pull request merged**